test/built-ins/TypedArray/prototype/slice/detached-buffer-get-ctor.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.slice
+description: Throws a TypeError buffer is detached on Get constructor.
+info: >
+  22.2.3.24 %TypedArray%.prototype.slice ( start, end )
+
+  ...
+  9. Let A be ? TypedArraySpeciesCreate(O, « count »).
+  ...
+  14. If SameValue(srcType, targetType) is false, then
+    ...
+  15. Else if count > 0, then
+    a. Let srcBuffer be the value of O's [[ViewedArrayBuffer]] internal slot.
+    b. If IsDetachedBuffer(srcBuffer) is true, throw a TypeError exception.
+  ...
+includes: [testTypedArray.js, detachArrayBuffer.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA();
+
+  Object.defineProperty(sample, "constructor", {
+    get: function() {
+      $DETACHBUFFER(sample.buffer);
+    }
+  });
+  assert.throws(TypeError, function() {
+    sample.slice();
```

Comment:
I believe this test is incorrect:

- step 3: len is 0
- step 4: relativeStart is 0
- step 5: k is 0
- step 6: relativeEnd is 0
- step 7: final is 0
- step 8: count is 0
- step 15: "count > 0" is false

...meaning step 15.b is never executed, so there should be no TypeError.